Lib/collections/__init__.py

```diff
@@ -1010,9 +1011,8 @@ def __len__(self):
         return len(set().union(*self.maps))     # reuses stored hash values if possible
 
     def __iter__(self):
-        d = {}
-        for mapping in reversed(self.maps):
-            d.update(dict.fromkeys(mapping))    # reuses stored hash values if possible
+        lazy_views = (key_view.keys() for key_view in reversed(self.maps))
```

Comment:
The variables here are misnamed and don't reflect an understanding of what is being done.   The contents of `self.maps` are mappings, not views.  The views aren't created until the `keys()` method is called.  The views themselves aren't "lazy", they are a pass-through to the underlying dict.

This line should read:

`views = (mapping.keys() for mapping in reversed(self.maps))`

Also, note this is an API change.  We've tried to make minimal assumptions about the underlying mappings which may not be real dicts.  In particular, we haven't demanded that they support having views.